app/src/main/java/com/ventivader/ui/mainscreen/BLEConnectionViewModel.kt

```diff
@@ -23,14 +23,19 @@ class BLEConnectionViewModel(application: Application) : AndroidViewModel(applic
         // NJD TODO - listen for Notifications from the Service
     }
 
+    override fun onCleared() {
+        super.onCleared()
+
+        getApplication<VentivaderApplication>().bleConnectionManager.close()
+    }
+
     val solenoidParametersLiveData = MutableLiveData<SolenoidParameters>().apply {
         SolenoidParameters()
     }
 
-    fun sendSolenoidParameters() {
-        // TODO - GATT write operation
-        // For now, we will assume best effort.. eventually we would use blocking 'pending' modal and not
```

Comment:
Not sure I would have deleted this comment.

Reply:
I think this is by mistake.